commitrmnocb/outcome.go

```diff
@@ -333,3 +452,55 @@ func counts[T comparable](elems []T) map[T]int {
 
 	return m
 }
+
+// MedianTimestamp returns the middle timestamp after sorting the provided timestamps.
+// empty/default is returned if the provided slice is empty.
+func MedianTimestamp(timestamps []time.Time) time.Time {
+	if len(timestamps) == 0 {
+		return time.Time{}
+	}
+	valsCopy := make([]time.Time, len(timestamps))
+	copy(valsCopy[:], timestamps[:])
+	sort.Slice(valsCopy, func(i, j int) bool {
+		return valsCopy[i].Before(valsCopy[j])
+	})
+
+	return valsCopy[len(valsCopy)/2]
+}
+
+// Deviates checks if x1 and x2 deviates based on the provided ppb (parts per billion)
+// ppb is calculated based on the smaller value of the two
+// e.g, if x1 > x2, deviation_parts_per_billion = ((x1 - x2) / x2) * 1e9
+func Deviates(x1, x2 *big.Int, ppb int64) bool {
+	// if x1 == 0 or x2 == 0, deviates if x2 != x1, to avoid the relative division by 0 error
+	if x1.BitLen() == 0 || x2.BitLen() == 0 {
+		return x1.Cmp(x2) != 0
+	}
+	diff := big.NewInt(0).Sub(x1, x2) // diff = x1-x2
+	diff.Mul(diff, big.NewInt(1e9))   // diff = diff * 1e9
```

Comment:
Should 1e9 be a const?

Reply:
I think it's okay within this small function to be like that.